### DISCUSSION TITLE
How can we listen to the same presence across tabs?

### DISCUSSION THREAD

**Discussion by csalmeida** [supabase/supabase#13865] [Questions]
We have a realtime application that is working nicely but we are trying to key our presence to the ID of our user or a fingerprint we are generating for the browser.

The problem is, supabase realtime when joining generates a new presence_ref for each browser tab connecting. Our object looks like this:

```javascript
{
    "8c302bc1-aa36-45b5-9c06-693c1982143a": [
        {
            "id": "8c302bc1-aa36-45b5-9c06-693c1982143a",
            "name": "Rob",
            "presence_ref": "F1e1bZSjLirE95bG",
            "slide": 0
        },
        {
            "id": "8c302bc1-aa36-45b5-9c06-693c1982143a",
            "name": "Rob",
            "presence_ref": "F1e1cu2WgafE9x5l",
            "slide": 0
        }
    ]
}
```

Each of these users should basically stay in sync with each other, but we can not identify which browser tab relates to which presence_ref.

The ideal scenario is that we could control the presence_ref so we wouldn't actually end up with this duplication, but that doesn't seem possible.

Additionally, we can't even obtain the presence_ref from our investigation. It doesn't come back in the join event and it isn't returned from what we can see on the channel join.

Where are we going wrong? It's surely possible to know which join relates to which presence_ref?

**Comment by SasheVuchkov** [comment]
I plan to solve this issue by tracking my own unique id for every connected client
